lib/CodeTransform/Adapter/WorseReflection/Refactor/WorseSimplifyClassName.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Phpactor\CodeTransform\Adapter\WorseReflection\Refactor;
+
+use Microsoft\PhpParser\Node\QualifiedName;
+use Microsoft\PhpParser\Parser;
+use Phpactor\CodeBuilder\Domain\BuilderFactory;
+use Phpactor\CodeTransform\Domain\Refactor\SimplifyClassName;
+use Phpactor\CodeTransform\Domain\SourceCode;
+use Phpactor\TextDocument\TextDocumentEdits;
+use Phpactor\TextDocument\TextEdit;
+use Phpactor\TextDocument\TextEdits;
+use Phpactor\WorseReflection\Core\Type\ClassType;
+use Phpactor\WorseReflection\Reflector;
+
+class WorseSimplifyClassName implements SimplifyClassName
+{
+    private Reflector $reflector;
+
+    private BuilderFactory $factory;
+
+    private Parser $parser;
+
+    public function __construct(
+        Reflector $reflector,
+        BuilderFactory $sourceBuilder,
+        Parser $parser = null
+    ) {
+        $this->reflector = $reflector;
+        $this->factory = $sourceBuilder;
+        $this->parser = $parser ?: new Parser();
+    }
+
+    public function getTextEdits(SourceCode $sourceCode, int $offset): TextDocumentEdits
+    {
+        $symbolContext = $this->reflector
+            ->reflectOffset($sourceCode->__toString(), $offset)
+            ->symbolContext();
+        $type = $symbolContext->type();
+
+        if (!$type instanceof ClassType) {
+            return new TextDocumentEdits($sourceCode->uri(), TextEdits::none());
+        }
+
+        $sourceBuilder = $this->factory->fromSource($sourceCode);
+        $sourceBuilder->use((string) $type->name());
+        $sourceBuilder->build();
+
+        $newClassName = $type->name()->short();
+        $position = $symbolContext->symbol()->position();
+
+        return new TextDocumentEdits($sourceCode->uri(), TextEdits::fromTextEdits([
+            TextEdit::create($position->start(), $position->end() - $position->start(), $newClassName)
+        ]));
```

Comment:
I don't know how to merge the content of the `$sourceBuilder` and the text edit I made for updating the token with the FQN. Could you maybe give me a hint for that.

Reply:
you need to use the `Updater`, f.e. https://github.com/phpactor/phpactor/blob/07e39698b42f1a146341a8dd62eec7b76e8d7876/lib/CodeTransform/Adapter/TolerantParser/Refactor/TolerantImportName.php#L152

Reply:
Yeah of course. Thanks.